src/base/path.lua

```diff
@@ -230,7 +231,7 @@
 			return p
 		end
 
-		if not newext:findlast(".", true) then
+		if newext ~= "" and not newext:findlast(".", true) then
```

Comment:
if #newext > 0

would be a little faster then the string compare...
but I also think checking for 'nil' would be nice, since then you can write

path.replaceextension(s)

So I'd change this to:

if newext and #newext > 0 and not newext:findlast(".", true) then